### PR TITLE
Feat/user data

### PR DESCRIPTION
**Pull request by adith-a-danthi**
## Overview

- Add `user data context`
- Add screens for `Liked Videos`, `Watch Later`, and `History`.
- Implement logic and API calls for the same.

Test Credentials
- email: `guest@breadboard.com`
- password: `password`

### CODE REVIEW

**Review by AnuragSahu11 — APPROVED** [review]
Good use of destructuring , nice folder structure you can merge.

File: src/utils/history-utils.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import axios from 'axios';
+
+const getHistory = async (dispatchUserData) => {
+  try {
+    const response = await axios.get('/api/user/history', {
+      headers: { authorization: localStorage.getItem('token') },
```
Comment (Hamdrive) [inline src/utils/history-utils.js:6]: You can consider fetching token from local storage inside your auth context and passing it where needed

**Review by Hamdrive — COMMENTED** [review]
Good work implementing logic for all video library features and context
Have suggested alternatives but optional

File: src/utils/history-utils.js (hunk 1 above)
Comment (adith-a-danthi) [reply src/utils/history-utils.js:6]: Makes sense. Will do it once done with the rest.

**Pull request merged**